### DISCUSSION TITLE
VB.NET Full Language Specification?

### DISCUSSION THREAD

**Discussion by DualBrain** [CommunityVB/Main#6] [Ideas]
It has been brought to my attention that the [VB.NET Full Language Specification](https://github.com/dotnet/vblang) is significantly out-of-date.  Is there anyone in our community that might have the skill/knowledge to dig through the compiler in order to find the information necessary to bring this document up to date?

**Comment by AdamSpeight2008** [comment]
I have previously wanted to rewrite the specification into a simpler form, broken down into smaller sections [see here](https://github.com/dotnet/vblang/pull/178/files).
Each having simple "cut and paste examples", these would be tending toward console applications, as to reduce extraneous noise.

**Comment by DualBrain** [comment]
Just read the responses over on that thread... it really doesn't make a whole lot of sense to state that they are "resisting" reformat because they are "gonna do something" that clearly doesn't appear to have any forward momentum.  The documentation is becoming out of date and there really doesn't seem to be anyone keeping it up to date.  At least doing what you were thinking would allow *something* to be maintained and utilized.  Furthermore, the argument as stated over there seems to be a bit weak as there, at least to me, seems that there should be a way to have a format that could potentially work as source material for both print and web - furthermore, are all of the docs on Microsoft's website done in a manner that affords easily printing to a book (I think not).  So, I for one, am very interested in seeing some movement on this in *whatever form* - I feel it is more important to have the information in whatever format than to not have it at all. ;-)

**Comment by paul1956** [comment]
First just document what the compiler does, The current VB specification is ~15.0,  There are new features in 15.3, 15.5, 16 and now 16.9 some of which have NO documentation. Below is what was added from VB 15 to 16, 16.9 adds Property Init support. I have no idea what UnconstrainedTypeParameterInConditional is nor can I find any information on it, even looking through source I only see where an error is suppressed. 
```
                Case Feature.Tuples,
                    Feature.BinaryLiterals,
                    Feature.DigitSeparators
                    Return LanguageVersion.VisualBasic15

                Case Feature.InferredTupleNames
                    Return LanguageVersion.VisualBasic15_3

                Case Feature.LeadingDigitSeparator,
                    Feature.NonTrailingNamedArguments,
                    Feature.PrivateProtected
                    Return LanguageVersion.VisualBasic15_5

                Case Feature.UnconstrainedTypeParameterInConditional,
                    Feature.CommentsAfterLineContinuation
                    Return LanguageVersion.VisualBasic16
```

**Comment by AdamSpeight2008** [comment]
@paul1956 There is an example of `Feature.UnconstrainedTypeParameterInConditional` [here](https://github.com/dotnet/roslyn/issues/29874)

**Reply by paul1956** [reply]
Doesn't really help, it takes you to other related issues that are still open and it is not in language spec or any documentation.
